components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/osgi/LauncherActivator.java

```diff
@@ -43,18 +50,40 @@ public void start(BundleContext bundleContext) throws Exception {
     final String args = bundleContext.getProperty("nexus-args");
 
     MDC.put("userId", SYSTEM_USERID);
-    server = new Launcher(basePath, null, null, args.split(","));
-    server.startAsync();
+    launcher = new Launcher(basePath, null, null, args.split(","));
+    launcher.startAsync(
+        new Runnable()
+        {
+          @Override
+          public void run() {
+            connectorConfigurationTracker = new ConnectorConfigurationTracker(
+                bundleContext,
+                launcher.getServer()
+            );
+            connectorConfigurationTracker.open();
+          }
+        }
+    );
+
+    final Dictionary<String, ?> properties = new Hashtable<>(singletonMap("name", "nexus"));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Hashtable" by an unsynchronized one such as "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1149)